From a2452d03f31729e6e1f20ea59fa7fe3fa1b4848a Mon Sep 17 00:00:00 2001
From: Omar Ramirez Luna <omar.r.ramirez.luna@intel.com>
Date: Thu, 20 Oct 2016 11:39:42 -0700
Subject: [PATCH 22/27] ov8858: fix return logic on ov8858_s_ctrl

Return at the end of the function to give more readibility and
accomodate future additions to the flow.

Signed-off-by: Omar Ramirez Luna <omar.r.ramirez.luna@intel.com>
---
 drivers/media/i2c/ov8858.c | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/drivers/media/i2c/ov8858.c b/drivers/media/i2c/ov8858.c
index 3d792ae..8001995 100644
--- a/drivers/media/i2c/ov8858.c
+++ b/drivers/media/i2c/ov8858.c
@@ -1759,6 +1759,7 @@ static int ov8858_s_ctrl(struct v4l2_ctrl *ctrl)
 	struct ov8858_device *dev = container_of(
 		ctrl->handler, struct ov8858_device, ctrl_handler);
 	struct i2c_client *client = v4l2_get_subdevdata(&dev->sd);
+	int ret = 0;
 
 	/* input_lock is taken by the control framework, so it
 	 * doesn't need to be taken here.
@@ -1784,29 +1785,31 @@ static int ov8858_s_ctrl(struct v4l2_ctrl *ctrl)
 		dev->fmt_idx = 0;
 		dev->fps_index = 0;
 
-		return 0;
+		break;
 	case V4L2_CID_FOCUS_ABSOLUTE:
 		if (dev->vcm_driver && dev->vcm_driver->t_focus_abs)
-			return dev->vcm_driver->t_focus_abs(&dev->sd,
+			ret = dev->vcm_driver->t_focus_abs(&dev->sd,
 							    ctrl->val);
-		return 0;
+		break;
 	case V4L2_CID_EXPOSURE_AUTO_PRIORITY:
 		if (ctrl->val == V4L2_EXPOSURE_AUTO)
 			dev->limit_exposure_flag = false;
 		else if (ctrl->val == V4L2_EXPOSURE_APERTURE_PRIORITY)
 			dev->limit_exposure_flag = true;
-		return 0;
+		break;
 	case V4L2_CID_HFLIP:
 		dev->hflip = ctrl->val;
-		return 0;
+		break;
 	case V4L2_CID_VFLIP:
 		dev->vflip = ctrl->val;
-		return 0;
+		break;
 	default:
 		dev_err(&client->dev, "%s: Error: Invalid ctrl: 0x%X\n",
 			__func__, ctrl->id);
-		return -EINVAL;
+		ret = -EINVAL;
 	}
+
+	return ret;
 }
 
 static int ov8858_g_ctrl(struct v4l2_ctrl *ctrl)
-- 
2.9.3

